src/fluids.md

```diff
@@ -0,0 +1,184 @@
+velocity part which is also derived from \eqref{eq:bdf_short}. Consider
+
+\begin{align}
+    \label{eq:hlm}
+    \frac{b_0}{\Delta t} u^{n+1} - L(u^{n+1}) = -\nabla p^{n+1} + F^*(u^{n+1})
+\end{align}
+
+with the Dirichlet (essential type) boundary condition
+
+\begin{align}
+    u^{n+1} = g_D^{n+1}.
+\end{align}
+
+A detailed walk through can also be found in Franco et al (2020) [^2].
+
+**Note** *The notation is very similar to what is used in the code to make it easy
+to follow the theoretical explanation and understand what is done in the
+implementation.*
+
+[^1]: A. G. Tomboulides, J. C. Y. Lee & S. A. Orszag (1997) Numerical Simulation
+of Low Mach Number Reactive Flows
+
+[^2]: Michael Franco, Jean-Sylvain Camier, Julian Andrej, Will Pazner (2020)
+High-order matrix-free incompressible flow solvers with GPU acceleration and
+low-order refined preconditioners (https://arxiv.org/abs/1910.03032)
+
+## Boundary Conditions
+
+### Inflow and no-slip walls
+
+For inflow or no-slip wall boundary conditions one should use the method
+*NavierSolver::AddVelDirichletBC*. This enforces the value on $u^{n+1}$ in
+\eqref{eq:hlm}. It is valid to call this method multiple times on different
+boundary attributes of the mesh. The *NavierSolver* instance keeps track of the
+associated *Coefficient* and accompanying boundary attribute. The passed
+attribute array can be modified, deleted or reused, since a copy is created.
+
+### Pressure outlet
+
+If an outlet of a domain is supposed to represent a pressure outlet (e.g.
+zero-pressure), one should use the method *NavierSolver::AddPresDirichletBC*.
+This enforces the pressure value $p^{n+1}$ in \eqref{eq:prespois}.
+
+### Zero-stress
+
+This boundary condition is used to represent an outflow attribute. Due to the
+nature of the $H^1$ finite-element discretization, the terms arise naturally in
+\eqref{eq:prespois} and \eqref{eq:hlm} resulting in
+
+\begin{align}
+    \nu \nabla u \cdot \hat{n} - p \cdot \hat{n} = 0.
+\end{align}
+
+If there is no other boundary condition applied to a certain attribute, this
+boundary condition is applied automatically (not through modification but rather
+through the formulation).
+
+## Solvers and preconditioners
+
+The choice of solvers and preconditioners for \eqref{eq:prespois} and
+\eqref{eq:hlm} are essential for the performance and robustness of the
+simulation.
+
+The pressure Poisson equation \eqref{eq:prespois} is solved using the CG Krylov
+method in combination with the low-order refined preconditioning technique
+coupled with AMG (c.f. Franco et al (2020)[^2]).
+
+Due to the nature of the explicit time discretization of the nonlinear term, the
+method used is CFL (and therefore time step) bound. As a result the time
+derivative term in \eqref{eq:hlm} is dominating and a CG Krylov method
+preconditioned with Jacobi is sufficient.
+
+Depending on the problem, this results in the majority of time per time step is
```

Comment:
I think there's a grammatical error here.  One possibility is to change "is spent" with "being spent".